mk/lib.mk

```diff
@@ -56,7 +56,11 @@ else
 $(lib-libfile): $(objs)
 	@$(cmd-echo-silent) '  AR      $$@'
 	@mkdir -p $$(dir $$@)
+ifeq ($(strip $(libdeps-external)),)
 	$$(q)rm -f $$@ && $$(AR$(sm)) rcs $$@ $$^
+else
+	$$(q)rm -f $$@ && $$(AR$(sm)) rcT $$@ $$^ $(libdeps-external)
+endif
```

Comment:
Why is this needed? `link-ldadd += $(libdeps-external)` in `core/arch/arm/kernel/link.mk` should work equally well (perhaps with `--start-group`/`--end-group` for circular dependencies).